packages/holocron/src/promiseStore.jsx

```diff
@@ -0,0 +1,87 @@
+import { useDispatch } from 'react-redux';
+
+/** @typedef {import('redux-thunk').ThunkAction} ThunkAction */
+
+/**
+ * A class to store promises by domain and key
+ *
+ * In general, Module code should not interact with the promise store directly,
+ *   instead dispatching the below thunks, or using the provided hooks
+ */
+class PromiseStore {
+  constructor() {
+    this.promises = {};
+  }
+
+  /**
+   * @param {string} domain The domain of the promise, should be the module name
+   *   if being stored by a module, or the package name if being stored by a package
+   * @param {string} key The key of the promise, it is up to the caller to manage key duplication
+   * @param {Promise<any>} promise The promise to be stored
+   * @returns {void}
+   */
+  store = (domain, key, promise) => {
```

Comment:
```suggestion
  set = (domain, key, promise) => {
```

Reply:
I thought about this, I guess I then didn't articulate this, in code or description.

I was thinking about having the store function throw, or silently no-op if the promise already existed. There is no reason to even overwrite a promise, so a collision would indicate something has gone wrong.

Because of this, I didn't like `set` because i would expect `set` to both `store` and `update` depending if the promise already existed.